src/lib/integration_test_local_engine/mina_automation.ml

```diff
@@ -0,0 +1,462 @@
+open Core
+open Async
+open Currency
+open Signature_lib
+open Mina_base
+open Integration_test_lib
+
+let version = "3.9"
+
+module Network_config = struct
+  module Cli_inputs = Cli_inputs
+
+  type docker_volume_configs = {name: string; data: string}
+  [@@deriving to_yojson]
+
+  type block_producer_config =
+    { name: string
+    ; id: string
+    ; keypair: Network_keypair.t
+    ; public_key: string
+    ; private_key: string
+    ; keypair_secret: string
+    ; libp2p_secret: string }
+  [@@deriving to_yojson]
+
+  type snark_coordinator_configs =
+    {name: string; id: string; public_key: string; snark_worker_fee: string}
+  [@@deriving to_yojson]
+
+  type docker_config =
+    { version: string
+    ; stack_name: string
+    ; coda_image: string
+    ; docker_volume_configs: docker_volume_configs list
+    ; block_producer_configs: block_producer_config list
+    ; snark_coordinator_configs: snark_coordinator_configs list
+    ; log_precomputed_blocks: bool
+    ; archive_node_count: int
+    ; mina_archive_schema: string
+    ; runtime_config: Yojson.Safe.t
+          [@to_yojson fun j -> `String (Yojson.Safe.to_string j)] }
```

Comment:
Do you know how the resource allocation works? Maybe that needs to be configurable as well. I know that the default memory allocated by docker on mac is not sufficient to run a node.

Reply:
Good point! Will address in a follow up PR.